compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -50,7 +50,7 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
     (formal, span) => formal.argInfos match {
       case arg1 :: arg2 :: Nil if !defn.isBottomClass(arg2.typeSymbol) =>
         val tp1 = fullyDefinedType(arg1, "TypeTest argument", span)
-        val tp2 = fullyDefinedType(arg2, "TypeTest argument", span)
+        val tp2 = fullyDefinedType(arg2, "TypeTest argument", span).normalized
```

Comment:
That looks a bit suspicious. By default I would say that we need a really good reason to manually trigger type normalized. In principle all newly created types should already go thought nurmalization just after creation, and if fullyDefinedType creates non normalized types it might mean that there is a better spot in the code to do the normalization (deeper in the stack trace).

Note that we another place in the code where we do a `fullyDefinedType` call followed by a `.normalized` (which is equality suspicious):

https://github.com/lampepfl/dotty/blob/fe24fd3af2d985262b67394ac8d606a9830e0291/compiler/src/dotty/tools/dotc/typer/Synthesizer.scala#L163

My intition tells me that there is a way to both fix this issue and remove that `.normalized` in `Synthesizer`, and that this way would a better solution that the change done in this PR...

Reply:
Then it seems to be a general limitation. We should probably open a separate issue for this improvement.